packages/compiler-sfc/src/compileScript.ts

```diff
@@ -1716,11 +1716,7 @@ function inferRuntimeType(
 }
 
 function toRuntimeTypeString(types: string[]) {
-  return types.some(t => t === 'null')
-    ? `null`
```

Comment:
Why is the `null` check removed? I don't think it's related to the case this PR is fixing.

Reply:
```ts
// test cases
foo: ((item: any) => boolean) | null
expect(content).toMatch(`foo: { type: [Function, null], required: true }`)

The original processing logic will result in an error return `null` when i fix this 


```